statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+        integer giving the number of seasons, often it is 4 for quarterly data
+        or 12 for monthly data. Default is no seasonal effect.
+    trend : str{'n','c','t','ct'} or iterable, optional
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        Can be specified as a string where 'c' indicates a constant (i.e. a
+        degree zero component of the trend polynomial), 't' indicates a
+        linear trend with time, and 'ct' is both. Can also be specified as an
+        iterable defining the polynomial as in `numpy.poly1d`, where
+        `[1,1,0,1]` would denote :math:`a + bt + ct^3`. Default is to not
+        include a trend component.
+    measurement_error : boolean, optional
+        Whether or not to assume the observations were measured with error.
+        Default is False.
+    time_varying_regression : boolean, optional
+        Used when an exogenous dataset is provided to select whether or not
+        coefficients on the exogenous regressors are allowed to vary over time.
+        Default is False.
+    mle_regression : boolean, optional
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation or through the Kalman filter (i.e.
+        recursive least squares). If `time_varying_regression` is True, this
+        must be set to False. Default is True.
+    simple_differencing : boolean, optional
+        Whether or not to use conditional maximum likelihood estimation.
+        If True, differencing is performed prior to estimation, which discards
+        the first :math:`SD + d` initial rows but reuslts in a smaller
+        state-space formulation. If False, the full SARIMAX model is put in
+        state-space form so that all datapoints can be used in estimation.
+        Default is False.
+    enforce_stationarity : boolean, optional
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model. Default is True.
+    enforce_invertibility : boolean, optional
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model. Default is True.
+    hamilton_representation : boolean, optional
+        Whether or not to use the Hamilton representation of an ARMA process
+        (if True) or the Harvey representation (if False). Default is False.
+
+    Attributes
+    ----------
+    measurement_error : boolean
+        Whether or not to assume the observations were measured with error.
+    state_error : boolean
+        Whether or not the transition equation has an error component.
+    mle_regression : boolean
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation.
+    state_regression : boolean
+        Whether or not to use estimate the regression coefficients via the
+        Kalman filter, by extending the state space.
```

Comment:
It's not quite the opposite because they could both be False. `mle_regression` is only True when there actually are regressors and coefficients were estimated via the MLE approach. Similarly, `state_regression` is only True when there are actually regressors and coefficients were estimated as part of the state vector.